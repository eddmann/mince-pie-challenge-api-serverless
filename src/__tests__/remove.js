import handler from '../handlers/remove';

it('requires an authenticated user', async () => {
  const services = { getUserIdFromToken: () => Promise.resolve() };

  const response = parseResponse(
    await handler(services)({ headers: {}, pathParameters: { id: '1' } }, {}),
  );

  expect(response.statusCode).toBe(401);
});

it('requires a pie to be present', async () => {
  const services = {
    getUserIdFromToken: () => Promise.resolve('USER_ID'),
    getPie: () => Promise.resolve(),
  };

  const response = parseResponse(
    await handler(services)({ headers: { Authorization: 'TOKEN' }, pathParameters: { id: '1' } }, {}),
  );

  expect(response.statusCode).toBe(404);
});

it('requires the user to own the pie', async () => {
  const services = {
    getUserIdFromToken: () => Promise.resolve('USER_ID'),
    getPie: () => Promise.resolve({ Id: '1', UserId: 'ANOTHER_USER_ID' }),
  };

  const response = parseResponse(
    await handler(services)({ headers: { Authorization: 'TOKEN' }, pathParameters: { id: '1' } }, {}),
  );

  expect(response.statusCode).toBe(403);
});

it('removes the users pie', async () => {
  const services = {
    getUserIdFromToken: () => Promise.resolve('USER_ID'),
    getPie: () => Promise.resolve({ Id: '1', UserId: 'USER_ID' }),
    removePie: jest.fn(),
  };

  const response = parseResponse(
    await handler(services)({ headers: { Authorization: 'TOKEN' }, pathParameters: { id: '1' } }, {}),
  );

  expect(response.statusCode).toBe(204);
  expect(services.removePie).toHaveBeenCalledWith('1');
});
