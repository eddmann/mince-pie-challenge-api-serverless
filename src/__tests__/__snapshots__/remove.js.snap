// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removes the users pie 1`] = `
Object {
  "body": "",
  "headers": Object {
    "Access-Control-Allow-Credentials": "*",
    "Access-Control-Allow-Origin": "*",
  },
  "statusCode": 204,
}
`;

exports[`requires a pie to be present 1`] = `
Object {
  "body": "{\\"title\\":\\"Not Found\\",\\"detail\\":\\"Unable to find the specified mince pie.\\"}",
  "headers": Object {
    "Access-Control-Allow-Credentials": "*",
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "application/problem+json",
  },
  "statusCode": 404,
}
`;

exports[`requires an authenticated user 1`] = `
Object {
  "body": "{\\"title\\":\\"Unauthorized\\",\\"detail\\":\\"Service requires an authenticated user\\"}",
  "headers": Object {
    "Access-Control-Allow-Credentials": "*",
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "application/problem+json",
  },
  "statusCode": 401,
}
`;

exports[`requires the user to own the pie 1`] = `
Object {
  "body": "{\\"title\\":\\"Forbidden\\",\\"detail\\":\\"This mince pie does not belong to you.\\"}",
  "headers": Object {
    "Access-Control-Allow-Credentials": "*",
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "application/problem+json",
  },
  "statusCode": 403,
}
`;
